src/Microsoft.Health.Fhir.Core/Features/Persistence/ResourceWrapperExtended.cs

```diff
@@ -0,0 +1,34 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+namespace Microsoft.Health.Fhir.Core.Features.Persistence
+{
+    public class ResourceWrapperExtended
+    {
+        public ResourceWrapperExtended(
+            ResourceWrapper wrapper,
+            bool allowCreate,
+            bool keepHistory,
+            WeakETag weakETag,
+            bool requireETagOnUpdate)
+        {
+            Wrapper = wrapper;
```

Comment:
Low priority: I think we should add validations to avoid wrapper to be null. Not sure if weakETag can be null as well.

This is a low priority item, if you don't have cycles to fix it, I can fix it in another branch. Just leave a comment.

Reply:
Looking the code I've noticed that weakETag can be null.

Reply:
I would avoid using default value null for week tag. This brought us to not testing not null value anywhere. I prefer explicit values. People should make caution decision on passing null.